Index: util-linux-worktree/term-utils/agetty.c
===================================================================
--- util-linux-worktree.orig/term-utils/agetty.c
+++ util-linux-worktree/term-utils/agetty.c
@@ -3027,7 +3027,7 @@ static void load_credentials(struct opti
 	dir = ul_path_opendir(pc, NULL);
 	if (!dir) {
 		log_warn(_("failed to open credentials directory"));
-		return;
+		goto out;
 	}
 
 	while ((d = xreaddir(dir))) {
@@ -3040,4 +3040,6 @@ static void load_credentials(struct opti
 		}
 	}
 	closedir(dir);
+out:
+	ul_unref_path(pc);
 }
