Index: util-linux/sys-utils/lscpu-arm.c
===================================================================
--- util-linux.orig/sys-utils/lscpu-arm.c	2025-06-19 16:54:39.000000000 +0800
+++ util-linux/sys-utils/lscpu-arm.c	2025-06-19 17:27:52.072178100 +0800
@@ -311,7 +311,8 @@
     { 0x66, faraday_part, "Faraday" },
     { 0x69, intel_part,   "Intel" },
     { 0x6d, ms_part,      "Microsoft" },
-    { 0x70, ft_part,      "Phytium" },
+    /* The parsing logic of phytium CPU is wrong. Use the value in /proc/cpuinfo directly. */
+    { 0x70, NULL,      "Phytium" },
     { 0xc0, ampere_part,  "Ampere" },
     { -1,   unknown_part, "unknown" },
 };
